src/main/java/ch/wisv/areafiftylan/users/controller/UserProfileRestController.java

```diff
@@ -48,19 +48,29 @@
     }
 
     /**
-     * Add a profile to a user. An empty profile is created when a user is created, so this method
-     * fills the existing fields
+     * This method returns a boolean indicating whether an user is allowed to make the changes to
+     * their profile or not.
      *
-     * @param userId The userId of the user to which the profile needs to be added
-     * @param input  A representation of the profile
-     * @return The user with the new profile
+     * @param user The concerning user
+     * @param input The changes
+     * @return boolean whether they are allowed to edit their profile
      */
-    @PreAuthorize("@currentUserServiceImpl.canAccessUser(principal, #userId)")
-    @PostMapping("/{userId}/profile")
-    public ResponseEntity<?> addProfile(@PathVariable Long userId, @Validated @RequestBody ProfileDTO input) {
-        User user = userService.addProfile(userId, input);
+    private boolean checkAllowedToEditProfile(User user, ProfileDTO input) {
+        boolean isUserCheckedIn = rfidService.isOwnerLinked(user.getEmail());
+
+        if (!isUserCheckedIn) {
+            return true;
+        }
+
+        LocalDate currentBirthday = user.getProfile().getBirthday();
+        boolean isDateChanged = !currentBirthday.toString().equals(input.getBirthday().toString());
```

Comment:
Why would you do an equals on the `toString()` methods and not just the objects itself?

Reply:
Removed it and changed it to use the equals method